google/cloud/spanner_dbapi/connection.py

```diff
@@ -122,11 +119,6 @@ def connect(
     )
 
     instance = client.instance(instance_id)
-    if not instance.exists():
-        raise ValueError("instance '%s' does not exist." % instance_id)
-
     database = instance.database(database_id, pool=spanner.pool.BurstyPool())
-    if not database.exists():
-        raise ValueError("database '%s' does not exist." % database_id)
```

Comment:
Can't say I like it. If we'll not check the existence here, a lot of objects (instance, database, session, transaction - with all of their state attributes) will be initialized before something will fail, so we'll still be using resources. And it'll fail with Google-specific error `google.api_core.exceptions.NotFound`. PEP 249 lists its own exceptions (and the Python default exceptions), which must be raised instead of those project-specific, so this `NotFound` error gonna be confusing.

Reply:
@IlyaFaer I recall having similar idea when working on Backups, and Google's feedback was in favor of not making "optional" admin API calls unless required by the specification (this may have something to do with the billing and price reduction). I see nothing in PEP-249 that requires those checks, and the expected normal setup implies having both the instance and the database to exist a priori. Therefore the responsibility can be delegated to the user who is free to choose whether to run the checks or not - at their own expense.